pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -703,7 +700,7 @@ func resolveTask(
 	} else {
 		spec = pipelineTask.TaskSpec.TaskSpec
 	}
-	return spec, taskName, kind, err
+	return spec, taskName, kind, t.DeepCopy(), refSource, err
```

Comment:
Similar to the ResolvedTask object below, could this be simplified to `(*v1beta1.Task, *v1beta1.RefSource, error)`?

- task can come from t.TaskSpec
- taskName can come from t.ObjectMeta.Name
- kind can come from t.TypeMeta

Reply:
Should this also be a ResolvedTask? 🤔

Reply:
> Should this also be a ResolvedTask? 🤔

🤔 Yeah that's a great suggestion, I think we could refactor this part of code a bit